server/src/main/java/io/druid/segment/indexing/DataSchema.java

```diff
@@ -39,33 +44,72 @@
   private static final Logger log = new Logger(DataSchema.class);
 
   private final String dataSource;
-  private final InputRowParser parser;
+  private final Map<String, Object> parser;
   private final AggregatorFactory[] aggregators;
   private final GranularitySpec granularitySpec;
 
+  private final ObjectMapper jsonMapper;
+
   @JsonCreator
   public DataSchema(
       @JsonProperty("dataSource") String dataSource,
-      @JsonProperty("parser") InputRowParser parser,
+      @JsonProperty("parser") Map<String, Object> parser,
       @JsonProperty("metricsSpec") AggregatorFactory[] aggregators,
-      @JsonProperty("granularitySpec") GranularitySpec granularitySpec
+      @JsonProperty("granularitySpec") GranularitySpec granularitySpec,
+      @JacksonInject ObjectMapper jsonMapper
```

Comment:
The original setup of that stuff was intending that specifying `@Json` or specifying nothing to be equivalent.  If we wanted to make it really explicit which one we are getting, we can remove that though.

Fwiw, this was done because back then `ObjectMapper` could only map to one format.  I remember Tatu talking about wanting to make it possible to map to multiple different formats (smile, json, bson, etc.) from the same `ObjectMapper`.  The long-term intent was more to switch to leveraging that behavior than it was to force explicit labeling.